frappe/public/js/frappe/form/form.js

```diff
@@ -421,6 +422,17 @@ frappe.ui.form.Form = class FrappeForm {
 		}
 	}
 
+
+	// Adds un beforeunload Listener to trigger a "Leave Site" Dialog, when a dirty (unsaved) form is closed
+	set_dirty_listener() {
+		addEventListener('beforeunload', (event) => {
+			if (cur_frm.is_dirty()) {
+			event.preventDefault();
+			return event.returnValue = 'There are unsaved changes, are you sure you want to exit?';
```

Comment:
https://developer.mozilla.org/en-US/docs/Web/API/Window/beforeunload_event

The custom string for the popup is deprecated

Reply:
I'll leave it in for backwards compatability